### PR TITLE
[Task3] create Movie Screen

### PR DESCRIPTION
**Pull request by ando-andy**
## Related Tickets

- [#Ticket ID](https://docs.google.com/spreadsheets/d/1F5V2sIUMGej3Wrp8qYRscf0KaVSlccBmqkRx49_hUVo/edit#gid=935273790)

Create UI for Movies screen
 - List movies now playing using CollectionView
 - List popular movies using CollectionView
 - Title label
 - Search button

Create a UI action
 - Click movie cell to view detail of the selected movie
 - Click the Search button to go to Search screen
 - Click More to view all list

Fetch data from API and display to the UI

## What's this PR do ?
Init base API and models 

## Library
*(List gem, library third party add new)*
Alamofire:  https://github.com/Alamofire/Alamofire
WenchaoD / FSPagerView:  https://github.com/WenchaoD/FSPagerView

## Impacted Areas in Application
*(List features, api, models or services that this PR will affect)*

API document: https://developers.themoviedb.org/3/authentication/create-session

## Checklist
- [x] It was tested in local success ?
- [x] Updated rake task, environment variable at [here](https://???)
- [x] Fill link PR into ticket and the opposite
- [x] Note reason, scope of influence, solution into ticket
- [x] Validate UI/Model/API

## Notes
*(Other notes)*

### CODE REVIEW

File: MovieDB/Domain/Models/Movie.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import ObjectMapper
 
 struct Movie {
-    var id: Int
+    var id = 0
```
Comment (thanhlong12312) [inline MovieDB/Domain/Models/Movie.swift]:
```suggestion
    var id: Int
```

File: MovieDB/Scenes/Movie/Cell/CategoryCell.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+//  CategoryCell.swift
+//  MovieDB
+//
+//  Created by kazutaka.ando on 2019/06/19.
+//  Copyright © 2019 Kazando. All rights reserved.
+//
+
+import UIKit
+
+class CategoryCell: UITableViewCell, NibReusable { // swiftlint:disable:this final_class
```
Comment (dongsonbk58) [inline MovieDB/Scenes/Movie/Cell/CategoryCell.swift]: add final

File: MovieDB/Scenes/Movie/View/MoviesHeaderView.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+extension MoviesHeaderView: FSPagerViewDataSource, FSPagerViewDelegate {
+    public func numberOfItems(in pagerView: FSPagerView) -> Int {
+        return movieBannerList.count
+    }
+    
+    public func pagerView(_ pagerView: FSPagerView, cellForItemAt index: Int) -> FSPagerViewCell {
+        let cell = pagerView.dequeueReusableCell(withReuseIdentifier: "cell", at: index)
+        let url = URL(string: API.Urls.baseURLImage + self.movieBannerList[index].backdropPath)
+        cell.backgroundColor = .red
```
Comment (dongsonbk58) [inline MovieDB/Scenes/Movie/View/MoviesHeaderView.swift]: move to func setContentForCell() inside Cell

File: MovieDB/Scenes/Movie/Cell/CategoryCell.swift (hunk 2 above)
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/Cell/CategoryCell.swift]: add final

File: MovieDB/Scenes/Movie/Cell/MovieCell.swift [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+//  Copyright © 2019 Kazando. All rights reserved.
+//
+
+import UIKit
+
+final class MovieCell: UICollectionViewCell, NibReusable {
+    
+    @IBOutlet weak var moviePosterImage: UIImageView!
+    
+    func bindingCell(_ MovieViewModel: MovieViewModel?) {
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/Cell/MovieCell.swift]:
```suggestion
    func bindViewModel(_ viewModel: MovieViewModel?) {
```

File: MovieDB/Scenes/Movie/MoviesViewController.swift [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    // MARK: Properties
+    
+    var viewModel: MoviesViewModel!
+    
+    var moviesBanners = [Movie]() {
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/MoviesViewController.swift]: add private

File: MovieDB/Scenes/Movie/MoviesViewController.swift [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        didSet {
+            tableView.reloadData()
+        }
+    }
+    
+    var categoryList = [CategoryType]()
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/MoviesViewController.swift]: add private

File: MovieDB/Scenes/Movie/MoviesViewController.swift [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+// MARK: - TableView
+extension MoviesViewController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.categoryList.count
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/MoviesViewController.swift]: remove self

File: MovieDB/Scenes/Movie/MoviesViewController.swift [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(for: indexPath, cellType: CategoryCell.self)
+            .then {
+                $0.categoryLabel.text = self.categoryList[indexPath.row].categoryTitle
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/MoviesViewController.swift]: weak self

File: MovieDB/Scenes/Movie/View/MoviesHeaderView.swift [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+//
+
+import UIKit
+import FSPagerView
+
+final class MoviesHeaderView: UIView, NibLoadable {
+    
+    @IBOutlet weak var pagerView: FSPagerView!
+    
+    var movieBannerList: [Movie] = [] {
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/View/MoviesHeaderView.swift]: add private

File: MovieDB/Scenes/Movie/View/MoviesHeaderView.swift [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    public func pagerView(_ pagerView: FSPagerView, cellForItemAt index: Int) -> FSPagerViewCell {
+        let cell = pagerView.dequeueReusableCell(withReuseIdentifier: "cell", at: index)
+        let url = URL(string: API.Urls.baseURLImage + self.movieBannerList[index].backdropPath)
+        cell.backgroundColor = .red
+        cell.imageView?.do {
+            $0.sd_setImage(with: url, completed: nil)
+            $0.contentMode = .scaleAspectFill
+            $0.clipsToBounds = true
+        }
+//        cell.imageView?.image = UIImage(named: "1.jpg")
```
Comment (thanhlong12312) [inline MovieDB/Scenes/Movie/View/MoviesHeaderView.swift]: remove line

File: MovieDB/Domain/Models/Movie.swift (hunk 1 above)
Comment (ando-andy) [reply MovieDB/Domain/Models/Movie.swift]: I fixed before but it still remained. Anyway, Fixed🙇‍♂️

File: MovieDB/Scenes/Movie/Cell/CategoryCell.swift (hunk 2 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/Cell/CategoryCell.swift]: Added🙇‍♂️

Comment (ando-andy) [reply MovieDB/Scenes/Movie/Cell/CategoryCell.swift]: Added🙇‍♂️

File: MovieDB/Scenes/Movie/MoviesViewController.swift (hunk 5 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/MoviesViewController.swift]: Added🙇‍♂️

File: MovieDB/Scenes/Movie/MoviesViewController.swift (hunk 6 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/MoviesViewController.swift]: Added🙇‍♂️

File: MovieDB/Scenes/Movie/View/MoviesHeaderView.swift (hunk 10 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/View/MoviesHeaderView.swift]: Added🙇‍♂️

File: MovieDB/Scenes/Movie/View/MoviesHeaderView.swift (hunk 9 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/View/MoviesHeaderView.swift]: Added🙇‍♂️

File: MovieDB/Scenes/Movie/Cell/MovieCell.swift (hunk 4 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/Cell/MovieCell.swift]: Modified 🙇‍♂️

File: MovieDB/Scenes/Movie/MoviesViewController.swift (hunk 7 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/MoviesViewController.swift]: Removed🙇‍♂️

File: MovieDB/Scenes/Movie/MoviesViewController.swift (hunk 8 above)
Comment (ando-andy) [reply MovieDB/Scenes/Movie/MoviesViewController.swift]: Added🙇‍♂️

**Pull request closed without merging**